xdsl/printer.py

```diff
@@ -254,8 +261,9 @@ def print_block(self, block: Block, print_block_args: bool = True) -> None:
 
         self._indent += 1
         for op in block.ops:
-            self._print_new_line()
-            self.print_op(op)
+            if not (op.has_trait(IsTerminator) and not print_block_terminator):
+                self._print_new_line()
+                self.print_op(op)
```

Comment:
nit:
the multiple negatives is a bit confusing, how about something like this:
```suggestion
            if not print_block_terminator and op.has_trait(IsTerminator):
                # Do not print terminator if `print_block_terminator` is False
                continue
            self._print_new_line()
            self.print_op(op)
```

Reply:
Also reorders the checks to have the cheaper one first